usl_pipeline/usl_lib/usl_lib/readers/hdf5_readers.py

```diff
@@ -0,0 +1,29 @@
+import typing
+from typing import Dict
+
+import h5py
+import numpy
+import numpy.typing as npt
+
+
+def read_from_hdf5(
+    input_stream: typing.BinaryIO,
+    group_name: str = "features",
+) -> Dict[str, npt.NDArray[numpy.float32]]:
```

Comment:
Not all of the layers are going to be float32 (the masks will be ints)
I'd just leave it as a generic array.

Reply:
Done